### PR TITLE
Factor out common checks for set_data in various Image subclasses.

### PR DESCRIPTION
**Pull request by anntzer**
## PR checklist

- [ ] "closes #0000" is in the body of the PR description to [link the related issue](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue)
- [ ] new and changed code is [tested](https://matplotlib.org/devdocs/devel/testing.html)
- [ ] *Plotting related* features are demonstrated in an [example](https://matplotlib.org/devdocs/devel/documenting_mpl.html#writing-examples-and-tutorials)
- [ ] *New Features* and *API Changes* are noted with a [directive and release note](https://matplotlib.org/devdocs/devel/coding_guide.html#new-features-and-api-changes)
- [ ] Documentation complies with [general](https://matplotlib.org/devdocs/devel/documenting_mpl.html#writing-rest-pages) and [docstring](https://matplotlib.org/devdocs/devel/documenting_mpl.html#writing-docstrings) guidelines

### CODE REVIEW

File: lib/matplotlib/image.py [hunk 1]
```diff
@@ -688,6 +688,34 @@ def write_png(self, fname):
                           bytes=True, norm=True)
         PIL.Image.fromarray(im).save(fname, format="png")
 
+    @staticmethod
+    def _prepare_array(A):
+        # Common checks and typecasts on A for various Image subclasses.
```
Comment (timhoffm) [inline lib/matplotlib/image.py:693]: "prepare_array" is a bit vague.

```suggestion
    def _normalize_image_array(A):
        """
        Transform image-like input into a normalized array represenation,
        that is used as data format in Image subclasses.
        """
```

Comment (anntzer) [reply lib/matplotlib/image.py:693]: Sure.

**Review by timhoffm — APPROVED** [review]

File: lib/matplotlib/image.py (hunk 1 above)
Comment (timhoffm) [reply lib/matplotlib/image.py]: Did you intentionally not take the docstring?

Comment (anntzer) [reply lib/matplotlib/image.py]: oops, missed it.  reworded and included now.

**Review by ksunden — APPROVED** [review]

**Pull request merged**